akka-docs/src/main/paradox/scala/stream/stream-io.md

```diff
@@ -110,6 +110,15 @@ of the server, and `q` in the case of the client. This is implemented by @scala[
 and concatenating a `Source` with a single `BYE` element which will then be sent after the original source completed]@java[using a custom `GraphStage`
 which completes the stream once it encounters such command].
 
+### Using bidi as framing protocol.
+
+Element boundary may get lost when streaming through Tcp connections.
```

Comment:
I think this is a miss-statement, the thing is _there is no element boundary_ in a stream of byte chunks from an IO perspective, this is true for files as well as network protocols such as TCP - you always need to frame at the application level where you know what your data is.

I'd go for:
```
### Using framing in your protocol

Network connections just pass streams of bytes, and does not know what is a logical chunk of bytes from the application's point of view. Often when implementing network protocols you will want to introduce your own framing. This can be done in two ways:
```